crates/kumod/src/smtp_server.rs

```diff
@@ -555,6 +566,11 @@ pub struct GenericEsmtpListenerParams {
 
     #[serde(default)]
     line_length_hard_limit: Option<usize>,
+
+    #[serde(default)]
+    pub proxy_mode: Option<ProxyMode>,
```

Comment:
from a naming perspective, this might be better phrased as `allowed_proxy_mode` and possibly should be a list/set of `allowed_proxy_modes` so that we can subsequently support multiple options.